docs/Developers/Profile_Bears.rst

```diff
@@ -0,0 +1,75 @@
+Profile Bears
+=============
+
+This document provides an overview of coala's Profiling Interface. The
+Profiling Interface provides an abilities to profile Bear code to optimize its
+performance.
+
+The profiler will start by profiling the run() method of bears because this is
+the part where bear writers will spend time on, as rest of the part like loading
+the files, collecting the settings, etc. are done by coala itself.
+
+.. note::
+
+    Enabliling both profiler and debugger on a bear at a same time will result
+    into error.
+
+Bear writers will have the ability to directly dump the raw profile output
+(binary output) either on current working directory or to a specified directory
+name, which can be further used for examination of profiler stats with the help
+of different modules like ``pstats`` or ``snakeviz``.
+
+``--profile`` or ``profile`` (using coafile) is the main argument to enable
+profiling. It accepts an additional parameter directory name or directory path
+where profiled files will be dumped, if ``True`` is provided profiled files will
+be saved to current working directory. If specified directory does not exists it
+will create one and if path specified is already a file path then it will throw
+``FileExitsError``.
```

Comment:
I guess all this belongs into the `Usage` section. It should be maybe stripped as well since you describe already how to enable the profiler below. But the `FileExistsError` is important, that could go directly below the `Usage` paragraph, describing how paths are treated.